docs/reference/index-modules/allocation/delayed.asciidoc

```diff
@@ -0,0 +1,89 @@
+[[delayed-allocation]]
+=== Delaying allocation when a node leaves
+
+When a node leaves the cluster for whatever reason, intentional or otherwise,
+the master reacts by:
+
+* Promoting a replica shard to primary to replace any primaries that were on the node.
+* Allocating replica shards to replace the missing replicas (assuming there are enough nodes).
+* Rebalancing shards evenly across the remaining nodes.
+
+These actions are intended to protect the cluster against data loss by
+ensuring that every shard is fully replicated as soon as possible.
+
+This ``shard-shuffle'' can, however, put a lot of extra load on the cluster
+which may not be necessary if the missing node is likely to return soon.
+Imagine this scenario:
+
+* Node 5 loses network connectivity.
+* The master promotes a replica shard to primary for each primary that was on Node 5.
+* The master allocates new replicas to other nodes in the cluster.
+* Each new replica makes an entire copy of the primary shard across the network.
+* More shards are moved to different nodes to rebalance the cluster.
+* Node 5 returns after a few minutes.
+* The master rebalances the cluster by allocating shards to Node 5.
+
+If the master had just waited for a few minutes, then the missing shards could
+have been re-allocated to Node 5 with the minimum of network traffic.  This
+process would be even quicker for idle shards (shards not receiving indexing
+requests) which have been automatically <<indices-synced-flush,sync-flushed>>.
+
+The allocation of replica shards which become unassigned because a node has
+left can be delayed with the `index.unassigned.node_left.delayed_timeout`
+dynamic setting, which defaults to `0` (reassign shards immediately).
+
+This setting can be updated on a live index (or on all indices):
+
+[source,js]
+------------------------------
+PUT /_all/_settings
+{
+  "settings": {
+    "index.unassigned.node_left.delayed_timeout": "5m"
+  }
+}
+------------------------------
+// AUTOSENSE
+
+With delayed allocation enabled, the above scenario changes to look like this:
+
+* Node 5 loses network connectivity.
+* The master promotes a replica shard to primary for each primary that was on Node 5.
+* The master logs a message that allocation of unassigned shards has been delayed, and for how long.
+* The cluster remains yellow because there are unassigned replica shards.
+* Node 5 returns after a few minutes, before the `timeout` expires.
```

Comment:
cluster health also return the number of delayed unassigned shards, maybe mention it?